hazelcast/src/main/java/com/hazelcast/query/PagingPredicateAccessor.java

```diff
@@ -16,24 +16,38 @@
 
 package com.hazelcast.query;
 
+import com.hazelcast.query.impl.predicates.PagingPredicateImpl;
+import com.hazelcast.util.IterationType;
+
 import java.util.Map;
 
 /**
  * This class provides paging accessor of predicate.
+ * <p>
+ * This class is for internal use only and may be changed without any notice.
```

Comment:
I wanted to, but apparently it exists because of OSGi - we want to hide the `PagingPredicate` internal API but at the same time, we want it to be accessible from other OSGi bundles (the client bundle in this case). So it cannot be made internal.